### PR TITLE
50 2 frontend fetches mapping

### PR DESCRIPTION
**Pull request by bravint**
changelog:

add data for season overview page

### CODE REVIEW

File: src/client/components/styling/card.css [hunk 1]
```diff
@@ -1,114 +1,125 @@
...
 
 .podium-participant {
-  display: flex;
-  gap: 10px;
-  align-items: center;
+    display: grid;
+    grid-template-columns: 2rem 5rem 1fr;
+    gap: 10px;
+    align-items: center;
+}
```
Comment (nathanSamosa) [inline src/client/components/styling/card.css:77]: Bravin's worst nightmare

**Review by nathanSamosa — APPROVED** [review]

**Review by polarbear23 — APPROVED** [review]

**Review by molloyconor511 — APPROVED** [review]

File: src/client/components/Competition.jsx [hunk 2]
```diff
@@ -1,97 +1,92 @@
...
-
-            const response = await fetch(API_URL.COMPETITION_GET_BY_ID, {
-                method: HTTP_METHOD.GET,
-                headers: {
-                    Authorization: token,
-                },
-            });
-
-            const result = await response.json();
+        const { seasons } = competitions[0];
```
[the remark this answers is not in the document]
Comment (bravint) [reply src/client/components/Competition.jsx:22]: I need to figure out how the id is passed down, might have to be through the URL, or via a prop. We can go through the various approaches, its an easy change once we come to a consensus.

File: src/client/components/Competition.jsx [hunk 3]
```diff
@@ -1,97 +1,92 @@
...
 
-            console.log(result)
+        setCurrentSeason(seasons.slice(seasons.length - 1));
```
Comment (vherus) [inline src/client/components/Competition.jsx:24]: What happens if the seasons array is empty? `seasons.slice(-1)` might have some interesting results

File: src/client/components/card/SeasonPodium.jsx [hunk 4]
```diff
@@ -1,22 +1,86 @@
...
+    }, [season]);
+
+    //     for each participant, get position per round
+
+    const generateDataForRounds = (standing) =>
+        rounds.map((round) => round.placements.filter((placement) => placement.participantId === standing.participant.id));
+
+    return (
+        <>
+            {seasonStandings.length > 0 && (
```
Comment (vherus) [inline src/client/components/card/SeasonPodium.jsx:59]: You're safe to treat length like a bool here, `seasonStandings.length && (` would work fine

Comment (vherus) [inline src/client/components/card/SeasonPodium.jsx:59]: You're safe to treat length like a bool here, `seasonStandings.length && (` would work fine

**Review by vherus — APPROVED** [review]

**Pull request merged**